### ISSUE TITLE
Too many warnings when running tests

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#6241]
Run of the test suite now throw over 3000 warnings.

```
1916 passed, 4 skipped, 5253 deselected, 225 xfailed, 3165 warnings in 451.50s (0:07:31) =
```

https://github.com/nilearn/nilearn/actions/runs/26301849886/job/77428767890#step:8:19594

It would be good to bring this number down in general.

**Comment by Remi-Gau** [comment]
Although some of those warnings are specific to the test suite, some of them are not and will be seen by users in their everyday code.

One of them is the one thrown by embedded maskers (in decoder, glm...) that throw a warning like this one at fit time

```
[NiftiMasker.fit] Generation of a mask has been requested (imgs != None) while a mask was given at masker creation. Given mask will be used.
``` 

It appears around 250 times in the tests and in several places in the example.

This warning is raised by niftimasker (instantiated with a mask) when fit is run with an input image, to let the user know that no 'implicit mask' will be computed from the input image and that the masker will fall back on the explicit mask passed at init.

When no input image is passed at fit time, the warning is not raised.

It kind of make sense to throw this warning when the user is using the masker to generate a mask

```
masker = NiftiMasker(mask_img=mask_img)
masker.fit(input_img)
mask = masker.mask_img_
```

But when doing fit_transform I am not sure there is a big added value to this warning

```
masker = NiftiMasker(mask_img=mask_img)
signal = masker.fit_transform(input_img)
```

Also in this case there is nothing the user can do to avoid this warning where as in the case above the user could avoid the warning by not passing an image to fit.

So for this one I am going to silence this warning when running fit_transform and also probably turn the warning into a RunTimeWarning because this feels more like we are letting the user know about something rather telling them that there is something wrong.

**Comment by Remi-Gau** [comment]
> also probably turn the warning into a RunTimeWarning because this feels more like we are letting the user know about something rather telling them that there is something wrong.

We may even consider turn that in an event to log only and not a warning

**Comment by Remi-Gau** [comment]
We are also getting a LOT of warnings from numpy due to joblib

D:\a\nilearn\nilearn\.tox\plotting\Lib\site-packages\joblib\numpy_pickle.py:207
Setting the shape on a NumPy array has been deprecated in NumPy 2.5. As an alternative, you can create a new view using np.reshape (with copy=False if needed). 

I believe this was fixed upstream (https://github.com/joblib/joblib/pull/1777) and there is nothing more for us to do

**Comment by Remi-Gau** [comment]
Getting a deprecation from matplotlib:

D:\a\nilearn\nilearn\nilearn\plotting\html_stat_map.py:444
The set_bad function will be deprecated in a future version. Use cmap.with_extremes(bad=...) or Colormap(bad=...) instead.

**Comment by saeedrafsharx** [comment]
Happy to help,

`largest_connected_component` in `nilearn/_utils/ndimage.py` does:

    if not volume.dtype.isnative:
        volume.dtype = volume.dtype.newbyteorder("N")

That's 3 of the warnings under NumPy 2.5, from
`test_largest_cc_img_non_native_endian_type`. `volume.view(...)` reinterprets
the same buffer, so `label()` gives the same result (checked locally, 
little-endian only).

The catch is that the current line mutates the caller's array in place and the
docstring Notes document that. `.view()` wouldn't. Fine to drop that, or would
you rather keep it byte-identical?

I Can pick up more of these if it's useful. Small PRs per module, or one sweep?

**Comment by bthirion** [comment]
IIUC, it would be possible to change the docstrings to indicate the mutation. Which is fine with me.
